src/s_sound.c

```diff
@@ -228,6 +230,28 @@ static int S_AdjustSoundParams(const mobj_t *listener, const mobj_t *source,
   return *vol > 0;
 }
 
+//
+// S_CompareChannels
+//
+// A comparison function that determines which sound channel should
+// take priority. Can be used with std::sort.
+//
+// Returns true if the first channel should precede the second.
+//
+static int S_CompareChannels(const void *arg_a, const void *arg_b)
+{
+  const channel_t *a = (const channel_t *) arg_a;
+  const channel_t *b = (const channel_t *) arg_b;
+
+  // Note that a higher priority number means lower priority!
+  const int ret = a->priority - b->priority;
+
+  return ret ? ret : (b->idnum - a->idnum);
+}
+
+// How many instances of the same sfx can be playing concurrently
+static const unsigned int max_instances = 3;
```

Comment:
Shouldn't this be configurable?

Reply:
Erm, yes, right. I'll add that tomorrow.